base/src/main/java/org/aya/tyck/error/TypeMismatchError.java

```diff
@@ -0,0 +1,42 @@
+package org.aya.tyck.error;
+
+import org.aya.api.distill.DistillerOptions;
+import org.aya.api.error.ExprProblem;
+import org.aya.api.util.NormalizeMode;
+import org.aya.concrete.Expr;
+import org.aya.core.term.Term;
+import org.aya.pretty.doc.Doc;
+import org.jetbrains.annotations.NotNull;
+
+// Difference from BadTypeError: desired type is a term rather than Doc
+// TODO: Maybe there is better ways to deal with this?
+public record TypeMismatchError(
```

Comment:
What about making `BadTypeError.desired` an `AyaDocile`?

Reply:
Great! But what about `Docile` instead of `AyaDocile`? `Doc` doesn't extend `AyaDocile`, which makes much original code incompatible. Besides, there isn't much difference to me anyway.

Reply:
No, I'm wrong. The problem seems just lies in the difference between `AyaDocile` and `Docile`.  `Doc.xxx` can't be used anymore if using `AyaDocile`, which breaks the original code :sob: .

Reply:
> `Doc.xxx` can't be used anymore if using `AyaDocile`

Maybe we should generate `Doc` from `AyaDocile` with `AyaDocile.toDoc` first

Reply:
And there's a simple way to make `Doc` act like an `AyaDocile`: replace `Doc.xxx()` with `options -> Doc.xxx()`

Reply:
Wow, that's amazing. I'll try that. Thanks a lot!